substrate/frame/offences/src/lib.rs

```diff
@@ -73,14 +73,28 @@ pub mod pallet {
 
 	/// The primary structure that holds all offence records keyed by report identifiers.
 	#[pallet::storage]
-	#[pallet::getter(fn reports)]
 	pub type Reports<T: Config> = StorageMap<
 		_,
 		Twox64Concat,
 		ReportIdOf<T>,
 		OffenceDetails<T::AccountId, T::IdentificationTuple>,
 	>;
 
+	impl<T: Config> Pallet<T> {
+		#[deprecated(note = "Use Reports::<T>::get(id) instead.")]
+		pub fn reports<KArg>(
+			k: KArg,
+		) -> Option<OffenceDetails<T::AccountId, T::IdentificationTuple>>
+		where
+			KArg: frame_support::__private::codec::EncodeLike<ReportIdOf<T>>,
+		{
+			<Reports<T> as frame_support::storage::StorageMap<
+				ReportIdOf<T>,
+				OffenceDetails<T::AccountId, T::IdentificationTuple>,
+			>>::get(k)
```

Comment:
```suggestion
			Reports::<T>::get(report_id)
```

You can simply use your own deprecation note here :)